Source/Libs/GtkSharp/Box.cs

```diff
@@ -0,0 +1,30 @@
+// Gtk.Box.cs - Gtk Box class customizations
+//
+// Author: Marcel Tiede
+//
+// Copyright (C) 2019 Marcel Tiede
+//
+// This program is free software; you can redistribute it and/or
+// modify it under the terms of version 2 of the Lesser GNU General 
+// Public License as published by the Free Software Foundation.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this program; if not, write to the
+// Free Software Foundation, Inc., 59 Temple Place - Suite 330,
+// Boston, MA 02111-1307, USA.
+
+namespace Gtk {
+
+	using System;
+	using System.Runtime.InteropServices;
+
+	public partial class Box {
+
+		protected Box() : base() { }
+	}
```

Comment:
why this change is needed?

the autogenerated code looked like this:

```
		protected Box() : base(IntPtr.Zero)
		{
			CreateNativeObject (new string [0], new GLib.Value [0]);
		}
```

Reply:
Don't know, taken from #90

Reply:
For `Box` protected constructor is not generated

Reply:
box again ...
before, there was no chance to subclass a Box without base constructor 

`public Box (Gtk.Orientation orientation, int spacing) : base (IntPtr.Zero)`

now, i can subclass like this:

```
        public class MyBox : Box {
            
        }
```

i don't know if this is intended ....

Reply:
Intended. Now you can create template for `Box`. So `Box` is root class in glade.

Reply:
Glade is not very friendly for Box, but you can create Box. Select Box in Containers popup, then right click in object tree and `Add widget as toplevel`.

Reply:
despite this...:

https://docs.gtk.org/gtk3/ctor.Box.new.html

Reply:
https://pygobject.readthedocs.io/en/latest/guide/gtk_template.html

In python they inherit from Box also for template example

Reply:
i tried MyBox in samples, and it seems to work without error with simple usage.
but i don't know if something wents wrong with more complex usages.

Should we investigate more???

Reply:
I think current behavior is good. Can be fixed later if something is wrong.

Reply:
with this implementation 
`gtk_box_new`
is never called.

Reply:
`gtk_box_new` just sets `orientation` and `spacing` properties but they already have initialized values. C# class `Object.CreateNativeObject` -> `g_object_newv`.

https://gitlab.gnome.org/GNOME/gtk/-/blob/gtk-3-24/gtk/gtkbox.c

```c
GtkWidget*
gtk_box_new (GtkOrientation orientation,
             gint           spacing)
{
  return g_object_new (GTK_TYPE_BOX,
                       "orientation", orientation,
                       "spacing",     spacing,
                       NULL);
}
```

```c
static void
gtk_box_init (GtkBox *box)
{
  GtkBoxPrivate *private;

  box->priv = gtk_box_get_instance_private (box);
  private = box->priv;

  gtk_widget_set_has_window (GTK_WIDGET (box), FALSE);

  private->orientation = GTK_ORIENTATION_HORIZONTAL;
  private->children = NULL;

  private->default_expand = FALSE;
  private->homogeneous = FALSE;
  private->spacing = 0;
  private->spacing_set = FALSE;
  ...
```

Reply:
i see ... so we should have no problems, as the calls are  identical in the end.